src/components/timers/Countdown.tsx

```diff
@@ -1,3 +1,101 @@
-const Countdown = () => null;
+import { useState } from "react";
+import type { ChangeEvent } from "react";
+import { calculateSeconds } from "../../utils/helpers";
+import Time from "../generic/DisplayTime";
+import { Buttons } from "../generic/Button";
+import type { buttonInfo } from "../generic/Button";
+
+const interval = { intervalId: 0, isRunning: false };
+const initialState = { seconds: 0, minutes: 0 };
+
+const Countdown = () => {
+  const [currSeconds, setSeconds] = useState(0);
+  const setInputTime = (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.id === "numMinutes") {
+      initialState.minutes = event.target.valueAsNumber;
+    } else {
+      initialState.seconds = event.target.valueAsNumber;
+    }
+    setSeconds(calculateSeconds(initialState));
+  };
+  const decrementSeconds = () => {
+    setSeconds((currSeconds) => {
+      if (currSeconds > 0) {
+        return currSeconds - 1;
+      } else {
+        clearInterval(interval.intervalId);
+        interval.isRunning = false;
+        return currSeconds;
+      }
+    });
+  };
+  const onStart = () => {
+    if (!interval.isRunning && currSeconds > 0) {
+      interval.intervalId = setInterval(decrementSeconds, 1000);
```

Comment:
what happens if your component unmounts? you will be left with an interval running forever.